components/suggestion/switch_channel_provider.jsx

```diff
@@ -25,22 +25,42 @@ import {
 } from 'mattermost-redux/selectors/entities/users';
 import * as ChannelActions from 'mattermost-redux/actions/channels';
 
+import {sortChannelsByTypeAndDisplayName} from 'mattermost-redux/utils/channel_utils';
+
 import DraftIcon from 'components/svg/draft_icon';
 import GlobeIcon from 'components/svg/globe_icon';
 import LockIcon from 'components/svg/lock_icon';
 import ArchiveIcon from 'components/svg/archive_icon';
 import AppDispatcher from 'dispatcher/app_dispatcher.jsx';
 import {getPostDraft} from 'selectors/rhs';
 import store from 'stores/redux_store.jsx';
-import {getChannelDisplayName, sortChannelsByDisplayName} from 'utils/channel_utils.jsx';
 import {ActionTypes, Constants, StoragePrefixes} from 'utils/constants.jsx';
 import * as Utils from 'utils/utils.jsx';
+import UserStore from 'stores/user_store.jsx';
 
 import Provider from './provider.jsx';
 import Suggestion from './suggestion.jsx';
 
 const getState = store.getState;
 
+function getChannelDisplayName(channel) {
+    if (channel.type !== Constants.GM_CHANNEL) {
+        return channel.display_name;
+    }
+
+    const currentUser = UserStore.getCurrentUser();
```

Comment:
Adding a flux store use? You can import the redux store and use `store.getState()` if needed

Reply:
Nope just copied it out of another place so I could remove all the flux store usage from there. This file has a separate ticket for flux store removal.